verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -116,6 +116,18 @@ void IndexingFactsTreeExtractor::Visit(const SyntaxTreeNode& node) {
           ExtractClassInstances(node, register_variables);
           break;
         }
+
+        // for primitive types inside tagged with kRegisterVariable;
+        ExtractPrimitiveVariables(node, register_variables);
+        break;
+      }
+
+      const std::vector<TreeSearchMatch> variable_declaration_assign =
+          FindAllVariableDeclarationAssignment(node);
+
+      if (!variable_declaration_assign.empty()) {
+        ExtractPrimitiveVariables(node, variable_declaration_assign);
+        break;
```

Comment:
Does this nave to be done using FindAll... or could you leverage the visitation that is already happening?

Reply:
I think it can be leveraged.

Reply:
`kVariableDeclarationAssignment` is contained inside a `kDataDeclaration` Node so it will enter the switch case for `kDataDeclaration` and the traversal would stop. so I think it's okay to leave it this way.

Reply:
ok for now, but consider defaulting to traversing more subnodes might simplify some logic down the road.  It might reduce the number of cases you have to handle explicitly, but add a little more bookkeeping work to the extractor class if the information you need isn't already available in the Context().

Reply:
Okay got that.